cylc/flow/config.py

```diff
@@ -247,13 +242,12 @@ def __init__(
         self.mem_log("config.py:config.py: start init config")
         self.workflow = workflow  # workflow id
         self.workflow_name = get_workflow_name_from_id(self.workflow)
-        self.fpath = str(fpath)  # workflow definition
-        self.fdir = os.path.dirname(fpath)
-        self.run_dir = run_dir or get_workflow_run_dir(self.workflow)
-        self.log_dir = (log_dir or
-                        get_workflow_run_scheduler_log_dir(self.workflow))
-        self.share_dir = share_dir or get_workflow_run_share_dir(self.workflow)
-        self.work_dir = work_dir or get_workflow_run_work_dir(self.workflow)
+        self.fpath: Path = Path(fpath)  # workflow definition
```

Comment:
Would it be unreasonable to ask you for a docstring for the init method? I'd particularly like context for the non-keyword arguments, which I feel are more obscure/vague/generalized than the kwargs.

Reply:
TBH, for this piece of work, yes.

Reply:
Context for the non-keyword args:

* `workflow: str` is the workflow ID
* `fpath: Path` is the workflow file path
* `options` are the cli options

Reply:
Even though it is unreasonable 😁  I've added one with basic positional arg docs.